charms/reactive/flags.py

```diff
@@ -141,31 +147,34 @@ def toggle_flag(flag, should_set):
 
 @cmdline.subcommand()
 @cmdline.no_output
-def register_trigger(when=None, when_not=None, set_flag=None, clear_flag=None):
+def register_trigger(when=None, when_not=None, set_flag=None, clear_flag=None, callback=None):
     """
     Register a trigger to set or clear a flag when a given flag is set.
 
     Note: Flag triggers are handled at the same time that the given flag is set.
 
     :param str when: Flag to trigger on when it is set.
     :param str when_not: Flag to trigger on when it is cleared.
-    :param str set_flag: If given, this flag will be set when `when` is set.
-    :param str clear_flag: If given, this flag will be cleared when `when` is set.
+    :param str set_flag: If given, this flag will be set when the relevant flag is changed.
+    :param str clear_flag: If given, this flag will be cleared when the relevant flag is changed.
+    :param str callback: If given, this callback will be invoked when the relevant flag is changed.
```

Comment:
is callback a `str`?  It think it's and `Optional[Callable[]]` or something like that?

Reply:
Right.  Just expected to be a `lambda` expression or similar.  See example in above reply.